docs/platform_ccl.rst

```diff
@@ -0,0 +1,297 @@
+
+.. code-block::
+   :linenos:
+
+	"meas_units" :
+	{
+	  "count": 2,
+	  "connection_map":
+	  {
+	    "0" : [0, 2, 3, 5, 6],
+	    "1" : [1, 4]
+	  }
+	},
+
+
+.. _fig_qubit_numbering_ccl:
+
+.. figure:: ./qubit_number.png
+    :width: 800px
+    :align: center
+    :alt: Qubit numbering and connectivity of qubits in CC-Light Platform
+    :figclass: align-center
+
+    Qubit numbering and connectivity in CC-Light Platform.
+
+
+For the details below, it will be convinient to consider the 
+Figure :numref:`fig_qubit_numbering_ccl`, which shows qubit and edge numbering in CC-Light platform.
+
+
+Edges and the constraints imposed by these edges are specified in ``edges`` section.
+``count`` at Line 3 specifies the number of edges between qubits in the platform.
+``connection_map`` specifies connections. For example, on Line 6, Edge 0 implies a
+constraint on Edge 2 and Edge 10. This means, if Edge 0 is reserved for an operation,
+an operation on Edge 2 and Edge 10 will not be scheduled, until operation on Edge 0
+is complete.
+
+.. code-block::
+   :linenos:
+
+	"edges":
+	{
+	  "count": 16,
+	  "connection_map":
+	  {
+	    "0": [2, 10], 
+	    "1": [3, 11],
+	    "2": [0, 8],
+	    "3": [1, 9],
+	    "4": [6, 14],
+	    "5": [7, 15],
+	    "6": [4, 12],
+	    "7": [5, 13],
+	    "8": [2, 10],
+	    "9": [3, 11],
+	    "10": [0, 8],
+	    "11": [1, 9],
+	    "12": [6, 14],
+	    "13": [7, 15],
+	    "14": [4, 12],
+	    "15": [5, 13]
+	  }
+	},
+
+
+Detunning constraints are specified ``detuned_qubits``. A two-qubit flux gate
+lowers the frequency of its source qubit to get near the frequency of its target
+qubit.  Any two qubits which have near frequencies execute a two-qubit flux gate.
+To prevent any neighbor qubit of the source qubit that has the same frequency as
+the target qubit to interact as well, those neighbors must have their frequency
+detuned (lowered out of the way).  A detuned qubit cannot execute a single-qubit
+rotation (an instruction of 'mw' type).  An edge is a pair of qubits which can
+execute a two-qubit flux gate.  There are 'count' qubits. For each edge it is
+described, when executing a two-qubit gate for it, which set of qubits it detunes.
+
+.. code-block::
+   :linenos:
+
+	"detuned_qubits":
+	{
+	    "count": 7,
+	    "connection_map":
+	    {
+	    "0": [3],
+	    "1": [2],
+	    "2": [4],
+	    "3": [3],
+	    "4": [],
+	    "5": [6],
+	    "6": [5],
+	    "7": [],
+	    "8": [3],
+	    "9": [2],
+	    "10": [4],
+	    "11": [3],
+	    "12": [],
+	    "13": [6],
+	    "14": [5],
+	    "15": []
+	    }
+	}
+
+Qubit topology is specified/configured using ``topology`` section. This information is used in mapping of quantum circuit. This section starts with the specification of x and y demensions of grid by setting ``x_size`` and ``y_size``. Next, the available qubits in the platform are given an ``id``, and for each qubit its ``x`` and ``y`` position on the grid is specified.
```

Comment:
`This information is used in mapping of quantum circuit.` does this mean that OpenQL may attempt to insert SWAP operations to map circuits whose connectivity does not match that of the target platform?
If so, that should be clearly indicated, if not, it should be specified what the "mapping of circuits" means and how it can influence the output circuit generated from OpenQL!

Reply:
yes, mapper may insert swap when enabled. will be detailed in mapping sections